dropwizard-jersey/src/main/java/io/dropwizard/jersey/errors/EofExceptionWriterInterceptor.java

```diff
@@ -0,0 +1,32 @@
+package io.dropwizard.jersey.errors;
+
+import org.eclipse.jetty.io.EofException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.ext.Provider;
+import javax.ws.rs.ext.WriterInterceptor;
+import javax.ws.rs.ext.WriterInterceptorContext;
+import java.io.IOException;
+
+/**
+ * A {@link WriterInterceptor} to swallow {@link org.eclipse.jetty.io.EofException} which occurs when a client
+ * disconnects before the complete response could be sent.
+ *
+ * @see EarlyEofExceptionMapper
+ * @see EofException
+ */
+@Provider
+public class EofExceptionWriterInterceptor implements WriterInterceptor {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EofExceptionWriterInterceptor.class);
+
+    @Override
+    public void aroundWriteTo(WriterInterceptorContext context) throws IOException, WebApplicationException {
+        try {
+            context.proceed();
+        } catch (EofException e) {
+            LOGGER.debug("Client disconnected while processing and sending response", e);
```

Comment:
We could also add a metric for this if it's regarded as useful.

Reply:
Having this as a metric could be useful indeed. I can imagine that a higher frequency of these exceptions might be an indicator for an overloaded HTTP server.